applylib/applyset/results.go

```diff
@@ -113,13 +113,14 @@ func (r *ApplyResults) pruneSuccess(gvk schema.GroupVersionKind, nn types.Namesp
 }
 
 // reportHealth records the health of an object.
-func (r *ApplyResults) reportHealth(gvk schema.GroupVersionKind, nn types.NamespacedName, isHealthy bool, message string) {
+func (r *ApplyResults) reportHealth(gvk schema.GroupVersionKind, nn types.NamespacedName, isHealthy bool, message string, err error) {
 	r.Objects = append(r.Objects, ObjectStatus{
 		GVK:           gvk,
 		NameNamespace: nn,
 		IsHealthy:     isHealthy,
 		IsPruned:      false,
 		Message:       message,
+		Error:         err,
```

Comment:
This should be something like HealthError I think, so that it's clear that the error applies to health computation.  Maybe we need to replace `IsHealthy bool`  with `Health HealthInfo`, where HealthInfo is something like `struct { Healthy bool; Message string; Error err; }`